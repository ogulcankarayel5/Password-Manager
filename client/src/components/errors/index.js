export * from './StatusError';
export * from './styles'