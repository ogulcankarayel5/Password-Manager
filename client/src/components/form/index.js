//export * from './Login';
//export * from './Register';
export * from './FormHoc';
export * from './FormInput'
export * from './SocialButtonComponent'
export * from './styles'