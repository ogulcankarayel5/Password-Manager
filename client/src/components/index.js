
export * from './header';
export * from './layout';
export * from './mainSection';
export * from './footer';

export * from './errors';
export * from './form';
export * from './seo'