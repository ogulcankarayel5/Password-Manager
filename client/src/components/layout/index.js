export * from './CommonLayout';
export * from './FormLayout';