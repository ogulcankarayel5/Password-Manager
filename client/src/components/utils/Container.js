import styled from "styled-components";
import { device } from "../../utils/breakPoints";

export const Container = styled.div`
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;

  @media ${device.mobile} {
    max-width: 540px;
  }
  @media ${device.tablet} {
    max-width: 720px;
  }
  @media ${device.desktop} {
    max-width: 960px;
  }
  @media ${device.large} {
    max-width: 1140px;
  }
`;

