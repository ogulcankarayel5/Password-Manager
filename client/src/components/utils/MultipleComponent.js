import styled from "styled-components";
import {Link} from 'react-router-dom';

export const Title = styled.h1`
  font-size: 6rem;
  color: ${(props) => props.theme.colors.mainTitle};
`;

export const Text = styled.p`
  font-size: 2rem;
  color: ${(props) => props.theme.colors.text};
  font-weight: 600;
  line-height:1.5;
`;

export const Button = styled(Link)`
  font-size:${props=>props.fontSize ?? '2rem'};
  color:${props=>props.theme.colors.background};
  background-color:${props=>props.theme.colors.primary};
  padding:${props=>props.padding ?? '2rem'};
  border-radius: 25%/50%;
  font-weight:${props=>props.fontWeight ?? 500};
  

`;

