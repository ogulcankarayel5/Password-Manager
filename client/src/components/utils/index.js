export * from './Container';
export * from './MultipleComponent';
export * from './Theme';

