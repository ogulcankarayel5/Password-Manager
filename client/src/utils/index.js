export * from './breakPoints';
export * from './globalStyle';
export * from './store';
export * from './history';
export * from './setAuthToken';
export * from './validate'
export * from './tests'
export * from './mockStore'
export * from './api'
export * from './actionType'
