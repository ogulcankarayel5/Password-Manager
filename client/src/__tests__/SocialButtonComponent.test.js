import  React from 'react';
import {shallow} from 'enzyme';
import {findByTestAtrr,checkProps} from '../utils';
import {SocialButtonComponent} from '../components'

describe("SocialButtonComponent",() => {

    describe("Checking PropTypes",() => {
        it("Should not throw a warning",() => {

            const expectedProps = {
                background:"red",
                color:"blue",
                borderColor:"green",
                spanText:"button component",
                onClick:()=>{},
                type:"submit"
            }
            const propsError = checkProps(SocialButtonComponent,expectedProps)
            expect(propsError).toBeUndefined();


        })
    })

    describe("Renders",() => {

        let wrapper;
        beforeEach(() => {

            const props = {
                background:"red",
                color:"blue",
                borderColor:"green",
                spanText:"button component",
                onClick:()=>{},
                type:"submit"
            }
            wrapper=shallow(<SocialButtonComponent {...props}/>);
        })

        it("Should render a button",() => {
            const button = findByTestAtrr(wrapper,"buttonComponent");
            expect(button.length).toBe(1)
        })
    })
})