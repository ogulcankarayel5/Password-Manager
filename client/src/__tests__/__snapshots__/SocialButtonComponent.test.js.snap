// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="sc-fzoXzr kotTAm"
  color="red"
  data-test="buttonComponent"
  type="button"
>
  button
</button>
`;
